unified_frameworks/sensor_array/gps_compass/actual_gps_compass.py

```diff
@@ -40,22 +36,48 @@ def __init__(self) -> None:
 
         self.compass = LSM303.Compass()
 
+        self.gpsThreadCall = Thread(target=self.read)
+        self.gpsThreadCall.start()
+
+
+    def isGPSStateValid(self) -> bool:
+        """
+        Returns our GPS's current state.
+        """
+        return self.gpsState
+
     def get_cur_angle(self) -> float:
         return self.compass.get_heading()
 
-    def get_cur_gps(self) -> Tuple[int, int]:
+    def read(self) -> None:
+        """
+        On thread for reading gps coordinates
+        This way we can continously retrieve the latest data
+        """
         temp = self.gps.get_position()
```

Comment:
This should be in a while loop. As is, the thread only runs this function once and then exits.